We only validate the certificate if tls_opt_require_cert is set, and if we're
not validating the certificate, then the host name check provides no benefit.

Note: in OpenLDAP 2.0 the LDAP_OPT_X_TLS option appears to be ignored
completely in favor of the LDAP_OPT_X_TLS_REQUIRE_CERT option, while the
reverse is the case in OpenLDAP 2.1.

--- openldap-2.0.27/libraries/libldap/tls.c	2002-01-26 14:55:02.000000000 -0500
+++ openldap-2.0.27/libraries/libldap/tls.c	2004-05-26 17:47:22.000000000 -0400
@@ -1118,10 +1118,11 @@
 	/* 
 	 * compare host with name(s) in certificate
 	 */
-
-	ld->ld_errno = ldap_pvt_tls_check_hostname( ssl, host );
-	if (ld->ld_errno != LDAP_SUCCESS) {
-		return ld->ld_errno;
+	if ( tls_opt_require_cert ) {
+		ld->ld_errno = ldap_pvt_tls_check_hostname( ssl, host );
+		if (ld->ld_errno != LDAP_SUCCESS) {
+			return ld->ld_errno;
+		}
 	}
 
 	/*
